WIP: Remove orphaned strings and history events

Entropy512 opened the pull request:
Actual changes as part of #6457 
So far I've just done history events 1-99, starting work on the rest.
No history event from "Profile Loaded" looks fishy to me, is this real?  Is that referenced in some unusual way in the code?

Lawrence37 commented on rtengine/procevents.h:
```diff
@@ -89,7 +89,7 @@ enum ProcEventCode {
     EvCACorr = 64,
     EvHREnabled = 65,
-    obsolete_66 = 66, //obsolete
+    obsolete_66 = 66, // obsolete
```
The corresponding refresh map entry is still there. Can you remove it?

Entropy512 replied:
Good catch, it's a bit harder to find refreshmap entries that are historically orphaned.

Will move EVHRAmount to category 0 and mark as obsolete tonight

Entropy512 replied:
Done in the latest commit.

Lawrence37 commented on rtengine/procevents.h:
```diff
@@ -24,7 +24,7 @@ namespace rtengine
 // Aligned so the first entry starts on line 30
 enum ProcEventCode {
     EvPhotoLoaded = 0,
```
While you're revising this file, could you add a few blank lines to bump this one down to line 30 like the comment says?

There's also line 203 in refreshmap.cc which should be removed to preserve the alignment.

Entropy512 replied:
Hmm, before I do that, we should figure out *why* it starts on line 30.  Perhaps someone attempted to do an auto-parser script as a companion to generateUnusedKeys, similar to my findorphans?

No such script currently exists (and I see a decent amount of evidence it never did...), it may be better to simply delete the comment and have a script that can handle the start line moving around and/or commented lines.

As to line 203, that needs some investigation.  It looks like EvCATEnabled was moved from ALLNORAW to LUMINANCECURVE - looks like @Desmis did that in 5abd92209 - so that appears to be recent and intentional, @Desmis  is there a reason the old value was left in there but commented, as opposed to just changing it?

heckflosse replied:
> we should figure out why it starts on line 30

I did this long time ago to make it easier to (manually) check for consistency between procevents.h and refreshmap.cc
For example get an entry in procevents.h like `EvClip = 12,`. If the corresponding entry in refreshmap.cc is not in line 30+12 = 42, something is wrong.

Entropy512 replied:
Longterm we might want to figure out a way to automate that a bit.  I'll work on fixing the alignment for now as libclang's comments support is...  Hard to understand to say the least so automating that task is probably a post-5.9 thing.

Thanks for the information!

Entropy512 replied:
So I just looked at this - the comment is on line 28, but the first entry already is on line 30, no action needed there.

Lawrence37 replied:
The one in refreshmap.cc starts on 30 but this one is on line 26.

Entropy512 replied:
Aha, missed that.

Fixed.  I amended the last commit and forcepushed since this one was so small.

Lawrence37 commented:
> No history event from "Profile Loaded" looks fishy to me, is this real?

Sounds like an event for when the image is first loaded and the default profile is applied. I don't recall the history message being used, but I will dig into the code when I have time.

Desmis commented:
Hello all
I come to add my grain of salt in the debate, not that I contest in any way, the simplification, the wish of clarification, etc.
But we have to be very careful, otherwise RT will not work properly anymore.

There are 2 ways to proceed with the "events
* either the one mentioned with the correspondence between "procevents.h" and "refreshmap.cc" and you have to be very careful 
* or to use a single function incorporated in the GUI, for example as we find in "wavelet.cc" (there are many other cases...), where we declare in a single "message", "events", the level of this "event" which will condition the execution order and the message which appears in the GUI.

Of course in both cases it is necessary to check that the system works, especially from "improcoordinator.cc".

And as a reminder, I am not a good programmer at all, but I had a lot of difficulties to understand and make the system work, especially when you want to change the process order :)

jacques

Lawrence37 commented:
I could not find any references to the "Profile loaded" event or history message, even on a very old version of RawTherapee. There's only one place where the history message key gets generated. It uses the event code, but `EvProfileLoaded` is never referenced directly or indirectly via a cast. I think it's safe to remove.

Entropy512 commented:
Hmm, where was it referenced numerically?  I'd like to take a look just to be sure, might turn out to be a possible failing of my analysis scripts.

Entropy512 commented:
Ah, so when you say "the key gets generated" I'm assuming you mean by something that is traversing the events table?  e.g. generated from the table but never used?

Lawrence37 commented:
> There's only one place where the history message key gets generated.

I should reword this. I mean there is one piece of code responsible for generating messages keys.

It does so by taking in an event enum. Since `EvProfileLoaded` is never used directly or from a cast, HISTORY_MSG_2 never gets generated.

Entropy512 commented:
I'll try to do a few more 100-message chunks tonight.  Hopefully will finish all orphans by the end of the weekend.

Entropy512 commented:
HISTOGRAM_TOOLTIP_RAW is listed as an orphan, this kinda looks strange to me that there isn't a tooltip for this...

Most orphaned strings that aren't history events are locallab stuff - not surprising given how much work was done on that since 5.8, and some of the "orphaned" strings have wording indicating that they were experimental so it's no surprise they were dropped.

Entropy512 commented:
Also, as you go further into the events table, instead of moving things to obsolete_NN, they are instead commented out in procevents.h

I'm thinking that maybe we want to rework all of these commits to use commenting-out, so there's better history of what it used to be?  Thoughts?

Lawrence37 commented:
I don't think it matters if we comment them or our change them to obsolete. It's not like anyone needs to know that they were. If they do, it's simple enough to do a `git blame`.

Entropy512 commented:
In that case I'll continue on with the current method unless @heckflosse , @Floessie , or @Thanatomanic disagree.
Planning on finishing the orphaned strings tonight, hopefully tomorrow I'll take a crack at running generateUnusedKeys as a sanity check and generateTranslationDiffs as a final cleanup phase.

Thanatomanic commented:
I agree with @Lawrence37: if it becomes necessary, things are easy to add back in again. In general, there is sooo much commented code in RT that is simply not useful anymore, nor will it likely become relevant in the future.

marter001 commented:
just to make sure: this proceeding will not find missing entries, right?
Would it be better to open new issues for this?

Entropy512 commented:
Do you mean something that is in the code and missing from `default`, or something that is in `default` but not other language files?

generateTranslationDiffs should handle the second case

marter001 commented:
first one: choosing profile "none" in colormanagement, in history is shown (none) but there is no entry for this in default. The same for (camera)

Entropy512 commented:
Hmm...  That might be a second phase after this round of cleanup.  It's probably best to open a separate issue for those.

Lawrence37 reviewed (approved):
Looks good to merge.

Entropy512 commented:
Technically there's sttill more to do, although I'm OK doing a second round PR for that.

Sorry I haven't been around as much as I thought to finish this, a bit of general burnout lately.

Entropy512 commented:
Apologies for taking so long to finish this.

I squashed the history events commits, since the last bit of orphan removal hasn't been reviewed yet I haven't squashed it.  I intend to squash prior to someone merging.

Also rebased on latest dev

Entropy512 commented:
Looks like the CI system is borked for MacOS - since I didn't touch anything related to what bombed.

Lawrence37 commented on rtdata/languages/default:
```diff
@@ -3604,11 +3538,9 @@ TP_RAW_3PASSBEST;3-pass (Markesteijn)
 TP_RAW_4PASS;3-pass+fast
 TP_RAW_AHD;AHD
 TP_RAW_AMAZE;AMaZE
-TP_RAW_AMAZEBILINEAR;AMaZE+Bilinear
```
`TP_RAW_AMAZEBILINEAR;AMaZE+Bilinear`
`TP_RAW_DCBBILINEAR;DCB+Bilinear`
`TP_RAW_RCDBILINEAR;RCD+Bilinear`
These three dual-demosaic keys are in use.

Entropy512 replied:
Re-added in next commit.

Lawrence37 commented:
`HISTORY_MSG_112` can be removed and `EvPdShrMaskToggled` is obsolete.

Beep6581 commented:
Hello, I would like to review `default` and it seems like this PR needs to be merged before that. @Lawrence37 could you do that if ready?

Entropy512 commented:
I'll leave it up to you and @Lawrence37 whether you want to handle the small remaining issues he found in a separate PR or not.

If you want me to handle them I'll tyr to take care of that tonight, sorry I've been a horrible procrastinator (AGAIN).

Lawrence37 commented:
@Beep6581 @Entropy512 One obsolete string is not a big problem. Feel free to merge.

Lawrence37 commented:
@Entropy512 Ok to squash and merge?

Entropy512 commented:
Yeah.  I feel like there's more to be done, but that could be in another PR.

Entropy512 commented:
Squashed and rebased, cleaned up the commit message a bit

Lawrence37 commented:
@Beep6581 Merged.
